cpp/examples/all-sky/rrtmgp_allsky.cpp

```diff
@@ -147,12 +227,33 @@ int main(int argc , char **argv) {
         rel(icol,ilay) = merge(rel_val, 0., lwp(icol,ilay) > 0.);
         rei(icol,ilay) = merge(rei_val, 0., iwp(icol,ilay) > 0.);
       });
+#ifdef RRTMGP_ENABLE_KOKKOS
+      Kokkos::parallel_for( MDRangeP<2>({0,0}, {nlay,ncol}) , KOKKOS_LAMBDA (int ilay, int icol) {
+        cloud_mask_k(icol,ilay) = p_lay_k(icol,ilay) > 100. * 100. && p_lay_k(icol,ilay) < 900. * 100. && mod(icol+1, 3) != 0;
+        // Ice and liquid will overlap in a few layers
+        lwp_k(icol,ilay) = merge(10.,  0., cloud_mask_k(icol,ilay) && t_lay_k(icol,ilay) > 263.);
+        iwp_k(icol,ilay) = merge(10.,  0., cloud_mask_k(icol,ilay) && t_lay_k(icol,ilay) < 273.);
+        rel_k(icol,ilay) = merge(rel_val, 0., lwp_k(icol,ilay) > 0.);
+        rei_k(icol,ilay) = merge(rei_val, 0., iwp_k(icol,ilay) > 0.);
+      });
+      {
+        conv::compare_yakl_to_kokkos(cloud_mask, cloud_mask_k);
```

Comment:
Why does cloud_mask not get compared in the below call to compare_all_yakl_to_kokkos?

Reply:
Because it's a different type, an array of bools, not reals.